src/main/java/org/broadinstitute/hellbender/engine/VariantWalker.java

```diff
@@ -51,9 +53,13 @@ protected final void onStartup() {
 
         // Need to discover the right codec for the driving source of variants manually, since we are
         // treating it specially (separate from the other sources of Features in our FeatureManager).
-        FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(drivingVariantFile);
-        if (codec.getFeatureType() == VariantContext.class) {
+        final FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(drivingVariantFile);
+        if (VariantContext.class.equals(codec.getFeatureType())) {
```

Comment:
All it would save us is 1 method dispatch (if that - the JIT compiler is smart enough to inline this). And the code would require a comment (and most lint tools will complain about == on object types). So I'm not happy to make this change.